addons/account/views/report_invoice.xml

```diff
@@ -99,55 +99,46 @@
-                                    <tr>
-                                        <td><span t-esc="amount_by_group[0] if len(o.tax_line_ids) > 1 else (o.tax_line_ids.tax_id.description or o.tax_line_ids.tax_id.name)"/></td>
+                    <div class="clearfix">
+                        <div id="total" class="row pull-right">
+                            <div>
+                                <table class="table-condensed" style="min-width: 250px;max-width: 400px;">
+                                    <tr class="border-black" style="border-bottom:1px solid #dddddd;">
+                                        <td><strong>Subtotal</strong></td>
                                         <td class="text-right">
-                                            <span t-esc="amount_by_group[1]" t-options='{"widget": "monetary", "display_currency": o.currency_id}'/>
+                                            <span t-field="o.amount_untaxed" t-options='{"widget": "monetary", "display_currency": o.currency_id}'/>
                                         </td>
                                     </tr>
-                                </t>
-                                <tr class="border-black">
-                                    <td><strong>Total</strong></td>
-                                    <td class="text-right">
-                                         <span t-field="o.amount_total" t-options='{"widget": "monetary", "display_currency": o.currency_id}'/>
-                                    </td>
-                                </tr>
-                            </table>
-                        </div>
-                    </div>
-                    <!-- DO NOT REMOVE THIS TABLE. MANDATORY IN SOME COUNTRIES -->
-                    <div class="row" t-if="len(o.tax_line_ids) > 0">
-                        <div class="col-xs-6">
-                            <table class="table table-condensed">
-                                <thead>
-                                    <tr>
-                                        <th>Tax</th>
-                                        <th class="text-right">Base</th>
-                                        <th class="text-right">Amount</th>
-                                    </tr>
-                                </thead>
-                                <tbody>
-                                    <tr t-foreach="o.tax_line_ids" t-as="t">
-                                        <td><span t-field="t.tax_id.description"/></td>
-                                        <td class="text-right">
-                                            <span t-field="t.base" t-options='{"widget": "monetary", "display_currency": o.currency_id}'/>
-                                        </td>
+                                    <t t-set="tax_groups_length" t-value="len(o._get_tax_amount_by_group())"/>
+                                    <t t-foreach="o._get_tax_amount_by_group()" t-as="amount_by_group">
```

Comment:
Don't call `_get_tax_amount_by_group()` multiple times. If you need to reuse this, store it in a variable like: 
`<t t-set="tax_groups" t-value="o._get_tax_amount_by_group()"/>`
`<t t-foreach="tax_groups" t-as="amount_by_group">`

and to retrieve the len: `len(tax_groups)`.

I also notice you never use `tax_groups_length`... is it an old stuff?